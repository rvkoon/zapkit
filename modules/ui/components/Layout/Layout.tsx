import { ReactNode } from "react";
import { Navigation } from "@Navigation";

export interface ILayoutProps {
  children: ReactNode;
  withNavigation?: boolean;
}

export function Layout({
  children,
  withNavigation = true,
}: ILayoutProps): JSX.Element {
  return (
    <div className="bg-primary-blue min-h-screen">
      <div className="bg-[#543969] rounded-[50%] rotate-[10deg] blur-[150px] w-[120vw] md:w-[100vw] h-[40vh] fixed top-[-20vh] left-[-50vw] z-0"></div>
      <div className="relative z-10">
        {withNavigation && <Navigation />}
        <main className="py-25">{children}</main>
      </div>
    </div>
  );
}
