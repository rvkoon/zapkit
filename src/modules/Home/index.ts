export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';