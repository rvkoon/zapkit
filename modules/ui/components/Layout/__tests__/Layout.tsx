import { Layout, ILayoutProps } from "../Layout";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";
import mockRouter from "next-router-mock";

jest.mock("next/router", () => require("next-router-mock"));
describe("Layout component", () => {
  const defaultProps: ILayoutProps = {
    children: <div></div>,
  };

  function renderComponent(props: ILayoutProps): RenderResult {
    return render(<Layout {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({ ...defaultProps });
    expect(container).toMatchSnapshot();
  });
});
