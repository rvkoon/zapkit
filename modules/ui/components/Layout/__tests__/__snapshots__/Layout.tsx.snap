// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render correctly 1`] = `
<div>
  <div
    class="bg-primary-blue min-h-screen"
  >
    <div
      class="bg-[#543969] rounded-[50%] rotate-[10deg] blur-[150px] w-[120vw] md:w-[100vw] h-[40vh] fixed top-[-20vh] left-[-50vw] z-0"
    />
    <div
      class="relative z-10"
    >
      <nav
        class="p-25 w-max flex flex-col fixed top-0 left-0"
      >
        <a
          class="text-white py-15 px-2 flex gap-15 opacity-50 font-bold"
          href="/"
        >
          Home
        </a>
        <a
          class="text-white py-15 px-2 flex gap-15 opacity-50 font-bold"
          href="/store"
        >
          Store
        </a>
        <a
          class="text-white py-15 px-2 flex gap-15 opacity-50 font-bold"
          href="/profile"
        >
          Profile
        </a>
      </nav>
      <main
        class="py-25"
      >
        <div />
      </main>
    </div>
  </div>
</div>
`;
